ui/src/logs/containers/LogsPage.tsx

```diff
@@ -74,34 +100,49 @@ class LogsPage extends PureComponent<Props> {
 
   private handleChooseTimerange = (timeRange: TimeRange) => {
     this.props.setTimeRange(timeRange)
+    this.props.executeHistogramQuery()
```

Comment:
do either of these need to be awaited, to prevent a race condition? i believe these are async functions? this is where the `Async` suffix has been handy as a convention.

Reply:
potentially but will probably need to handle this the same way that I am handling zoom moving forward.